apiserver/facades/agent/uniter/watcherrelationunit.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package uniter
+
+import (
+	"context"
+
+	"github.com/juju/names/v6"
+	"github.com/juju/worker/v4/catacomb"
+
+	"github.com/juju/juju/core/application"
+	corerelation "github.com/juju/juju/core/relation"
+	coreunit "github.com/juju/juju/core/unit"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/domain/relation"
+	internalerrors "github.com/juju/juju/internal/errors"
+)
+
+// relationUnitsWatcher watches changes in related units for a specific unit
+// and relation.  It handles lifecycle management and manages buffers for
+// internal event streaming, working as a proxy around the domain watcher to
+// enrich its events.
+type relationUnitsWatcher struct {
+	catacomb catacomb.Catacomb
+
+	relation RelationService
+
+	unitName     coreunit.Name
+	relationUUID corerelation.UUID
+
+	out chan watcher.RelationUnitsChange
+}
+
+// newRelationUnitsWatcher creates and starts a watcher for observing changes
+// in relation units for a given unit and relation.
```

Comment:
todo: update comment.

> `for observing changes in relation units for a given unit and relation.`

For a given unit/relation combo, there is only one relation unit. Please update to say how the other relation units are related to the combo.